gandalf.js

```diff
@@ -1,22 +1,22 @@
-/**
```

Comment:
SCRIPT-This function seems to have grown quite complex. Can we break it down further?